salt/_modules/vault.py

```diff
@@ -0,0 +1,858 @@
+                        'Writing LDAP group -> Policy mappings for "%s"', str(auth_method.path))
+                    local_config_groups = auth_method.extra_config["group_policy_map"]
+                    for key in local_config_groups:
+                        log.debug('LDAP Group ["%s"] -> Policies %s',
+                                  str(key), local_config_groups[key])
+
+                        client.auth.ldap.create_or_update_group(
+                            name=key,
+                            policies=local_config_groups[key]
+                        )
+
+                    # Clean up LDAP group mapping
+                    if ldap_groups:
+                        for group in ldap_groups:
+                            if group in {k.lower(): v for k, v in local_config_groups.items()}:
+                                log.debug(
+                                    'LDAP group mapping ["%s"] exists in configuration, no cleanup necessary', group)
+                            else:
+                                log.debug(
+                                    'LDAP group mapping ["%s"] does not exist in configuration, deleting...', group)
+                                client.auth.ldap.delete_group(name=group)
+                                log.debug(
+                                    'LDAP group mapping ["%s"] deleted.', group)
+                else:
+                    log.debug(
+                        'Auth method "%s" does not contain any extra configurations.', auth_method.type
+                    )
+
+            # Build return object
+            ret['changes']['old'] =[ob.type for ob in remote_methods]
+            ret['changes']['new'] = new_auth_methods or "No changes"
+
+        except Exception:
+            raise
+
+        log.info('Finished processing and configuring auth methods...')
+
+    def cleanup_auth_methods(self, client, remote_methods, local_methods, ret):
+        """Disabling any auth methods not present in the local config
+
+        Arguments:
+            client {hvac} -- hvac client
+            remote_methods {list} -- auth methods from remote vault server
+            local_methods {list} -- auth methods from local config
+            ret {dict} -- salt state result
+        """
+        log.info('Cleaning up auth methods...')
+        has_change = False
+
+        try:
+            for auth_method in remote_methods:
+                if auth_method not in local_methods:
+                    has_change = True
+                    log.debug(
+                        'Auth method "%s" does not exist in configuration. Disabling...', auth_method.type)
+                    client.sys.disable_auth_method(
+                        path=auth_method.path
+                    )
+                    log.debug('Auth method "%s" is disabled.', auth_method.type)
+
+            if has_change:
+                ret['changes']['new'] = [ob.type for ob in local_methods]
+        except Exception:
```

Comment:
Since we are not actually handling any exception, just re-raising, this try/except block is not needed. Can remove it and let things raise exceptions normally.